include/rocksdb/slice.h

```diff
@@ -116,6 +118,81 @@ class Slice {
   // Intentionally copyable
 };
 
+/**
+ * A Slice that can be pinned with some cleanup tasks, which will be run upon
+ * ::Reset() or object destruction, whichever is invoked first. This can be used
+ * to avoid memcpy by having the PinnsableSlice object referring to the data
+ * that is locked in the memory and release them after the data is consuned.
+ */
+class PinnableSlice : public Slice, public Cleanable {
```

Comment:
Google C++ Style doesn't encourage this: https://google.github.io/styleguide/cppguide.html#Multiple_Inheritance